Detour/Source/DetourNavMesh.cpp

```diff
@@ -518,7 +519,30 @@ void dtNavMesh::connectExtOffMeshLinks(dtMeshTile* tile, dtMeshTile* target, int
 			}
 		}
 	}
+}
 
+void dtNavMesh::connectExtNonNeighborOffMeshLinks(dtMeshTile *tile)
+{
+	static const int MAX_TILES = 32;
+	dtMeshTile* tiles[MAX_TILES];
+	int ntiles;
+	for (int i = 0; i < tile->header->offMeshConCount; ++i)
+	{
+		dtOffMeshConnection* selfCon = &tile->offMeshCons[i];
+		const float *targetPos = &selfCon->pos[3];
+		int tx, ty;
+		calcTileLoc(targetPos, &tx, &ty);
+		if (abs(tx - tile->header->x) <= 1 && abs(ty - tile->header->y) <= 1)
+		{
+			continue;
+		}
+		int ntiles = getTilesAt(tx, ty, tiles, MAX_TILES);
+		for (int j = 0; j < ntiles; ++j)
+		{
+			connectExtOffMeshLinks(tile, tiles[j], selfCon->side);
+			connectExtOffMeshLinks(tiles[j], tile, dtOppositeTile(selfCon->side));
```

Comment:
This does not work correctly. Connections are stored in the tile where their start points are.

If you have connection from tile (1,1) to tile (5,1), and first add tile (1,1) and then tile (5,1) the off-mesh link will not get connected, because tile (5,1) was not around when the first tile was added.

Similar issue arises when tiles (5,1) is removed before tile (1,1).

This is the reason why only neighbour tiles are considered for off-mesh links, as we can do the reciprocal add/remove as there aren't that many tiles to consider.

Reply:
@memononen thanks for your reply!
the `connectExtOffMeshLinks` function will connect all links from `target` to this `tile`, and it will be called both way. So in my opinion it is ok. The original use of `connectExtOffMeshLinks` between neighbors is already redundant.

what do you see?

Reply:
@memononen also, the newly-added function will look through all links and accurately find out which tiles will be linked to.
there is nothing different between connecting (1, 1) to (5, 1)  and connecting (1, 1) with its neighbors.

what i do is changing 'considering only neighbors' to 'considering all tiles which is calculated by link's position'

Reply:
however, what you say about 'tiles (5,1) is removed before tile (1,1)' is a problem... But it won't affect path-finding in that it will be considered when a link fails?

Reply:
@memononen fair point.

Reply:
@memononen hi, i have implemented that in my game and had it tested under dynamicly adding & removing tiles.
Would you approve this work-around? If you would i will migrate that to recast and test it in a few days.

Reply:
@memononen hi, would you have a look at this. Now timing problems are gone.
![image](https://user-images.githubusercontent.com/9924749/208284532-bd1cc193-32b3-4d6f-8d90-92bd49e9629c.png)

Reply:
also had it tested under adding & removing tiles 
![image](https://user-images.githubusercontent.com/9924749/208284650-f9191229-0099-4de1-bdbb-2a50980f1188.png)
![image](https://user-images.githubusercontent.com/9924749/208284669-e6992f07-fab1-47a2-9c2b-2d31bb664412.png)

Reply:
@grahamboree notice that you are assigned a similar issue.

Reply:
notice that STL is not approved.